software/sw_setup.mk

```diff
@@ -18,6 +18,6 @@ SRC+=$(BUILD_ESRC_DIR)/iob_cache_swreg.h $(BUILD_ESRC_DIR)/iob_cache_swreg_emb.c
 $(BUILD_ESRC_DIR)/iob_cache_swreg%: iob_cache_swreg%
```

Comment:
this will be repeated in every core for no reason. Solution:

lib should have hardware.mk, software.mk, simulation.mk, fpga.mk, etc

core should have hw_build.mk, sim_build.mk, fpga_build.mk

we already do this for setup with hw_setup.mk, sim_setup.mk, etc 

Please fix this asymmetry and help improve our build process

Reply:
Actually, the above is missing for setup and not build. However, it does not invalidate the need to do this for build.

We need to decide on the names:

libhw_setup.mk: includes hw_setup from core and is included by the makefiles below 

libsw_setup.mk: includes sw_setup from core and is included by the makefiles below

etc.

Reply:
Don't create everything now. The philosophy is set. Create libhw_setup.mk and  libsw_setup.mk which we need now.

Dont forget to test if mkregs.toml exists before running mkregs.py

Reply:
Update cache only : others will update when they need

Reply:
all this is a small job that should not take more than an hour to do